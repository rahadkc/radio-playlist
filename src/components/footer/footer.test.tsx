import { cleanup, render, screen } from '@testing-library/react'
import * as React from 'react'
import Footer from './index'

describe('<Footer> component', () => {
  afterEach(cleanup)

  const title = 'Test Title'
  const name = 'Test FM'

  test('should render Footer component', () => {
    render(<Footer title={title} />)

    // If Only 'Title' prop send nothing to display
    expect(screen.queryByText(title)).toBeNull()
    expect(screen.queryByText(title)).toMatchSnapshot()

    render(<Footer title={title} name={name} />)

    // If 'Name' prop send then display 'title' and 'name'
    expect(screen.getByText(title)).toBeInTheDocument()
    expect(screen.getByText(name)).toBeInTheDocument()
    expect(screen.getByText(name)).toMatchSnapshot()
  })
})
