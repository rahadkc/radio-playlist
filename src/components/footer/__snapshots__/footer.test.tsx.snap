// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> component should render Footer component 1`] = `null`;

exports[`<Footer> component should render Footer component 2`] = `
<div
  class="name"
>
  Test FM
</div>
`;
